src/exberry_int/integration_exberry.py

```diff
@@ -62,15 +64,21 @@ def integration_exberry_main(
     env: 'ExberryIntegrationEnv',
     events: 'IntegrationEvents'):
 
-    outbound_queue = asyncio.Queue()
+    outbound_queue = asyncio.PriorityQueue()
     session_started = asyncio.Event()
 
-    LOG.info(f"Preparing session...")
-    outbound_queue.put_nowait(make_order_book_depth())
-
-    async def enqueue_outbound(msg: dict):
+    async def enqueue_outbound(msg: dict, priority: Optional[int] = None):
+        global CURRENT_PRIORITY
         LOG.info(f"Enqueuing outbound message: {msg}")
-        await outbound_queue.put(msg)
+        if not priority:
+            CURRENT_PRIORITY = 1 if outbound_queue.qsize() == 0 else CURRENT_PRIORITY + 1
```

Comment:
I don't think you should ever need more than two priority levels - one normal priority level, and another higher priority level that is used to ensure that the first failing message is sent immediately after the connection is reopened.

(Within those priority levels, the priority queue should ensure that items are dequeued in the same order they were enqueued)